### PR TITLE
Make the Executor for PrometheusHttpServer configurable

### LINKED ISSUE #5243: PrometheusHttpServer FixedThreadPool
```
PrometheusHttpServer(String host, int port) {
    try {
      server = HttpServer.create(new InetSocketAddress(host, port), 3);
    } catch (IOException e) {
      throw new UncheckedIOException("Could not create Prometheus HTTP server", e);
    }
    MetricsHandler metricsHandler =
        new MetricsHandler(() -> getMetricProducer().collectAllMetrics());
    server.createContext("/", metricsHandler);
    server.createContext("/metrics", metricsHandler);
    server.createContext("/-/healthy", HealthHandler.INSTANCE);
    executor = Executors.newFixedThreadPool(5, THREAD_FACTORY);
    server.setExecutor(executor);
    start();
  }
```

Why is the number of threads fixed at 5？Can it be customized?

### PR DESCRIPTION
**Pull request by abhisesh**
Fixes #5243

### CODE REVIEW

File: exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerBuilder.java [hunk 1]
```diff
@@ -32,13 +38,30 @@ public PrometheusHttpServerBuilder setPort(int port) {
...
+  }
+
   /**
    * Returns a new {@link PrometheusHttpServer} with the configuration of this builder which can be
    * registered with a {@link io.opentelemetry.sdk.metrics.SdkMeterProvider}.
    */
   public PrometheusHttpServer build() {
-    return new PrometheusHttpServer(host, port);
+    ExecutorService executorService = this.executor;
+    if (executorService == null) {
```
Comment (mateuszrzeszutek) [inline exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerBuilder.java:54]: The `executorService` cannot be null here; you're already setting it in the default constructor. Consider removing the assignment in the constructor instead

File: exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerBuilder.java [hunk 2]
```diff
@@ -32,13 +38,30 @@ public PrometheusHttpServerBuilder setPort(int port) {
...
+      executorService = getDefaultExecutor();
+    }
+    return new PrometheusHttpServer(host, port, executorService);
+  }
+
+  PrometheusHttpServerBuilder() {
+    this.executor = getDefaultExecutor();
```
Comment (mateuszrzeszutek) [inline exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerBuilder.java:61]:
```suggestion
```

File: exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerTest.java [hunk 3]
```diff
@@ -295,6 +302,38 @@ void stringRepresentation() {
...
+            .setHost("localhost")
+            .setPort(port)
+            .setExecutor(scheduledExecutor)
+            .build();
+    assertThat(server)
+        .extracting(
+            "executor", as(InstanceOfAssertFactories.type(ScheduledThreadPoolExecutor.class)))
+        .satisfies(
+            executor -> {
+              assertThat(executor.getCorePoolSize()).isEqualTo(10);
```
Comment (mateuszrzeszutek) [inline exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerTest.java:333]: Nit: you can use
```suggestion
              assertThat(executor).isSameAs(scheduledExecutor);
```
for a stronger assertion

File: exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerTest.java [hunk 4]
```diff
@@ -295,6 +302,38 @@ void stringRepresentation() {
...
+  }
+
+  @Test
+  void customExecutor() throws IOException {
+    ScheduledExecutorService scheduledExecutor = Executors.newScheduledThreadPool(10);
+    int port;
+    try (ServerSocket socket2 = new ServerSocket(0)) {
+      port = socket2.getLocalPort();
+    }
+    PrometheusHttpServer server =
```
Comment (jack-berg) [inline exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerTest.java:322]: Can you wrap this in a `try / finally` and close the server after the test? Or try-with-resources: `try(PrometheusServer server = ...) { .. }`.

**Review by jack-berg — APPROVED** [review]
Just one small comment. Thanks!

File: exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerTest.java (hunk 4 above)
Comment (abhisesh) [reply exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerTest.java]: Done. Thanks for the feedback!

File: exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerBuilder.java (hunk 2 above)
Comment (abhisesh) [reply exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerBuilder.java]: Removed redundant stmt. Thanks for the feedback!

**Review by jkwatson — APPROVED** [review]

**Pull request merged**